### PR TITLE
ref(endpoints): OrganizationMemberWithRolesSerializer

### PR DESCRIPTION
**Pull request by mgaeta**
Combine logic into a BaseEndpoint.

### CODE REVIEW

**Comment by NisanthanNanthakumar** [comment]
@mgaeta needs a rebase

File: src/sentry/api/serializers/models/organization_member/expand/projects.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from typing import Any, Mapping, MutableMapping, Sequence, cast
```
Comment (NisanthanNanthakumar) [inline src/sentry/api/serializers/models/organization_member/expand/projects.py:1]: @mgaeta why put these serializers in another subdirectory? I think its ok to keep these files 1 level up since they're exported by `src/sentry/api/serializers/models/organization_member/__init__.py`

File: src/sentry/api/bases/organizationmember.py [hunk 2]
```diff
@@ -41,19 +52,25 @@ def convert_args(self, request: Request, organization_slug, member_id="me", *arg
...
-                Q(user__is_active=True) | Q(user__isnull=True),
-                organization=organization,
-                id=member_id,
-            )
+            try:
+                queryset = OrganizationMember.objects.filter(
+                    Q(user__is_active=True) | Q(user__isnull=True),
+                    organization=organization,
+                    id=member_id,
+                    invite_status=InviteStatus.APPROVED.value,
```
Comment (NisanthanNanthakumar) [inline src/sentry/api/bases/organizationmember.py:72]: @mgaeta this change to the query is separate to the reorganization. Can you move this change into a separate PR?

**Review by NisanthanNanthakumar — APPROVED** [review]
Looks good! small nit of moving a query change into a separate PR

**Review by JoshFerge — CHANGES_REQUESTED** [review]
Want to review this before it merges, taking a look now.

**Review by JoshFerge — CHANGES_REQUESTED** [review]
Overall I like the direction of this PR, and it does clean up parts of these endpoints + serializers. 

However, it is extremely hard to review. I've tried to review it by individual commit, but in some commits bugs were introduced and were fixed in later commits, so this doesn't work that well. 

There are also many endpoints changed, of which I'd want to verify test coverage/quality if I cannot follow along with the changes confidently.

Could you either modify your commits to contain self-contained correct behavior changes, or consider breaking up this PR into smaller chunks for readability?

**Comment by JoshFerge** [comment]
will also point to our guidelines here: https://develop.sentry.dev/code-review/#try-to-organize-your-work-in-a-way-that-makes-it-conducive-to-review

**Comment by JoshFerge** [comment]
thank you for splitting out. see my comments on the already merged https://github.com/getsentry/sentry/pull/32718, would like those addressed. I will look at this tomorrow.
